src/main/java/com/stripe/net/StripeRequest.java

```diff
@@ -104,37 +142,88 @@ public StripeRequest withAdditionalHeader(String name, String value) {
 
   private static URL buildURL(
       ApiResource.RequestMethod method, String spec, Map<String, Object> params)
-      throws IOException {
+      throws ApiConnectionException {
     StringBuilder sb = new StringBuilder();
 
     sb.append(spec);
 
-    URL specUrl = new URL(spec);
-    String specQueryString = specUrl.getQuery();
+    URL url = null;
+    try {
+      URL specUrl = new URL(spec);
+      String specQueryString = specUrl.getQuery();
 
-    if ((method != ApiResource.RequestMethod.POST) && (params != null)) {
-      String queryString = FormEncoder.createQueryString(params);
+      if ((method != ApiResource.RequestMethod.POST) && (params != null)) {
+        String queryString = FormEncoder.createQueryString(params);
 
-      if (queryString != null && !queryString.isEmpty()) {
-        if (specQueryString != null && !specQueryString.isEmpty()) {
-          sb.append("&");
-        } else {
-          sb.append("?");
+        if (queryString != null && !queryString.isEmpty()) {
+          if (specQueryString != null && !specQueryString.isEmpty()) {
+            sb.append("&");
+          } else {
+            sb.append("?");
+          }
+          sb.append(queryString);
         }
-        sb.append(queryString);
       }
-    }
 
-    return new URL(sb.toString());
+      url = new URL(sb.toString());
+    } catch (IOException e) {
+      handleIOException(e);
+    }
+    return url;
   }
 
   private static HttpContent buildContent(
-      ApiResource.RequestMethod method, Map<String, Object> params) throws IOException {
+      ApiResource.RequestMethod method,
+      Map<String, Object> params,
+      RawRequestOptions.ApiMode apiMode)
+      throws ApiConnectionException {
     if (method != ApiResource.RequestMethod.POST) {
       return null;
     }
 
-    return FormEncoder.createHttpContent(params);
+    HttpContent httpContent = null;
+    try {
+      if (apiMode == RawRequestOptions.ApiMode.PREVIEW) {
+        httpContent = JsonEncoder.createHttpContent(params);
```

Comment:
Is this ever used considering we take the string body?

Reply:
Ah good point, we don't need this switch in regular `buildContent`. I'll remove it